pkg/kubelet/qos/policy.go

```diff
@@ -80,3 +114,12 @@ func GetContainerOOMScoreAdjust(pod *v1.Pod, container *v1.Container, memoryCapa
 	}
 	return int(oomScoreAdjust)
 }
+
+func isSidecarContainer(pod *v1.Pod, container *v1.Container) bool {
+	for _, initContainer := range pod.Spec.InitContainers {
```

Comment:
you can only run this loop if the restartPolicy is configured for the container. Something like:

```
if container.RestartPolicy != nil && *container.RestartPolicy == v1.ContainerRestartPolicyAlways {
  // check that it is actually an init container
  for _, initContainer := range pod.Spec.InitContainers {
  if name matches, return true.
}
```

Reply:
Done